ledger/src/ui/src/bolos_ux.c

```diff
@@ -321,11 +322,8 @@ static void sample_main(void) {
                     break;
                 case RSK_PIN_CMD: // Send pin_buffer
                     reset_if_starting(RSK_META_CMD_UIOP);
-                    pin = APDU_AT(2);
-                    if ((pin >= 0) && (pin <= PIN_LENGTH)) {
-                        G_pin_buffer[pin] = APDU_AT(3);
-                        G_pin_buffer[pin + 1] = 0;
-                    }
+                    init_pin_ctx(&pin_ctx, G_pin_buffer);
+                    tx = update_pin_buffer(&pin_ctx);
```

Comment:
Just noticing this now: we're probably missing `rx` as a parameter, and corresponding validations (which are currently non-existent, but we should probably add).